File: Trip-Planner/src/app/classes/Trip.ts
import { Activiteit } from "./Activiteit";
import { UserTrip } from "./UserTrip";

export class Trip {
    tripId: number;
    startDatum?: Date | undefined;
    eindDatum: Date | null;
    lengte: number;
    naam: string;
    userTrip: UserTrip | null;
    activiteiten: Activiteit[] = [];
    userTrips: UserTrip[] = [];
    
    constructor(tripId: number, startDatum: Date, eindDatum: Date, lengte: number, userTrip: UserTrip, naam:string){
        this.tripId = tripId;
        this.startDatum = startDatum;
        this.eindDatum = eindDatum;
        this.naam = naam;
        this.lengte = lengte;
        this.userTrip = userTrip;
    }
}

File: TripPlannerAPI/TripPlannerDAL/Entity/Trip.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripPlannerDAL.Entity
{
    public class Trip
    {
        public int tripId {  get; set; }
        public string? naam { get; set; }
        public DateTime startDatum { get; set; }
        public DateTime eindDatum { get; set; }
        public int lengte { get; set; }
        public ICollection<Activiteit>? activiteiten { get; set; }
        public ICollection<UserTrip>? userTrips { get; set; }
    }
}
